### DISCUSSION TITLE
Windows on ARM version for AaxAudioConverter & BookLibConnect

### DISCUSSION THREAD

**Discussion by JustAUser467** [audiamus/AaxAudioConverter#216] [General]
Hello there,

I recently set up my new Surface Laptop 7 which comes with the Snapdragon cpu.
I tried to install both BookLibConnect and AaxAudioConverter with no luck. So I want to ask for a ARM-version of the two apps. Is there anything planed in the near future?
I only found a thread from a user who tried to set it up on a M1 Mac which is a year ago. Is there anything new on this topic?

Thanks in advance!

Best regards, David

**Comment by audiamus** [comment]
I have no experience with Windows on ARM so far, neither in emulation nor in native mode. But I'll put it on the todo list to further investigate.

**Reply by JustAUser467** [reply]
Inspired by your suggestion on the WIndows on ARM emulated on a Mac post I tried it the same way by copying the installed files from a x64 windows pc to my windows on ARM device. It just works!
After starting BookLib Connect the first time I was greeted by the message that Microsoft Runtime Desktop needs to be installed. I was linked to the website for downloading it and after installing it I successfully downloaded and converted a Audible audio book using your two applications.

**Comment by xecrets** [comment]
Hi, my forks at https://github.com/xecrets/BookLibConnect and https://github.com/xecrets/AaxAudioConverter have each a branch feature/arm64net8 that upgrades the projects to .NET 8, adds support for a publish profile for arm64 on Windows and fixes some minor issues that came up during the upgrades.

I've not done anything on the InnoSetup project, so that may not work. But the publish profile will produce a single framework dependent executable that works fine on arm64. Since I don't do the InnoSetup, you'll have to provide an ffmpeg build for arm64 at startup. I used https://github.com/tordona/ffmpeg-win-arm64 .

It's a quick & dirty upgrade, there are quite a few warnings that should be addressed etc.

I did have a conversation with the maintainer @audiamus a few years ago when I did an upgrade to .NET 7, but the plan then was to do a major overhaul and merge BookLibConnect with AaxAudioConverter so they declined that pull request.

I would be willing to do a little more solid work, either in my own fork, or with pull requests if the maintainer would say something about the future of the project. It's very, very useful so I'd hate to see it become abandoned.

Right now I don't publish any binaries or anything, but if you have Visual Studio 2022 it's trivial to download the source from my forks, select the feature branch, and hit "Publish". Then you get a single executable file to use. (It does require the .NET 8 runtime to be installed. Can't build self-contained with trimming because it does not work well with Windows Forms, so the result then becomes prohibitively large.)

I am running it on an UTM virtual windows guest on a Mac Book Air with Apple Silicon (arm64), and it runs very well and quite fast.

**Reply by JustAUser467** [reply]
Hi xecrets, thank you for your work on supporting Windows on ARM for those two applications.
After I had to reset my computer with a clean install of Windows 11 I tried to install the two packages you provide on the ARM64 branches. Unfortunatelly both won't install due to an error mentioning that this version of the application only runs on x64 Windows.
It happens to both BookLibConnect 0.13.1 and AaxAudioConverter 1.18.2.
Is there any fix available to get the applications running on my Windows on ARM machine?

Thanks in advance!

**Reply by xecrets** [reply]
Hello @JustAUser467 !

You need to build specifically the `feature/arm64net8` branch from https://github.com/xecrets/BookLibConnect and https://github.com/xecrets/AaxAudioConverter with Visual Studio 2026 (probably also ok to build with VS Code but I haven't tried it).

You also need to provide an ffmpeg build for arm64 at startup. I used https://github.com/tordona/ffmpeg-win-arm64 .

There are no pre-built packages to install, follow the instructions in my original comment and you should be fine. I rebuilt just a few weeks ago in order to fix a bug, which is also committed to the `feature/arm64net8` branch.

You also need to use the ARM64 publish profile, which is located in `Properties/PublishProfiles`:

<img width="1083" height="375" alt="Screenshot 2026-07-28 094750" src="https://github.com/user-attachments/assets/7b7653c1-c9eb-4999-8e97-b6fd3df145a6" />
